sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4058,6 +4058,44 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
         Row(1, 2, 1, 2) :: Nil)
     }
   }
+
+  test("SPARK-36093: RemoveRedundantAliases should not change expression's name") {
+    withTable("t1", "t2") {
+      withView("t1_v") {
+        sql("CREATE TABLE t1(cal_dt DATE) USING PARQUET")
+        sql(
+          """
+            |INSERT INTO t1 VALUES
+            |(date'2021-06-27'),
+            |(date'2021-06-28'),
+            |(date'2021-06-29'),
+            |(date'2021-06-30')""".stripMargin)
+        sql("CREATE VIEW t1_v AS SELECT * FROM t1")
+        sql(
+          """
+            |CREATE TABLE t2
+            |USING PARQUET
+            |PARTITIONED BY (CAL_DT)
+            |AS
+            |SELECT 1 AS FLAG,CAL_DT FROM t1_v
+            |WHERE CAL_DT BETWEEN '2021-06-27' AND '2021-06-28'""".stripMargin)
```

Comment:
so the first CTAS works fine before this PR? can we add `checkAnswer...` here?

Reply:
> so the first CTAS works fine before this PR? can we add `checkAnswer...` here?

Yes， since it use DataSource and store partition column as Seq[String], when executing generate InsertIntoHadoopFsRelation  use resolver. So the partitionColumns is not changed.

Reply:
> so the first CTAS works fine before this PR? can we add `checkAnswer...` here?

Update the `checkAnswer`

Reply:
if CTAS is fine, can we use `CREATE TABLE` here and just test the `INSERT`? to make the test simpler.

Reply:
> if CTAS is fine, can we use `CREATE TABLE` here and just test the `INSERT`? to make the test simpler.

Done